xds/internal/xdsclient/xdsresource/unmarshal_cds.go

```diff
@@ -177,6 +177,9 @@ func validateClusterAndConstructClusterUpdate(cluster *v3clusterpb.Cluster) (Clu
 		if err := proto.Unmarshal(cluster.GetClusterType().GetTypedConfig().GetValue(), clusters); err != nil {
 			return ClusterUpdate{}, fmt.Errorf("failed to unmarshal resource: %v", err)
 		}
+		if len(clusters.Clusters) == 0 {
+			return ClusterUpdate{}, fmt.Errorf("aggregate cluster has clusterConfig with empty Clusters field in response: %+v", cluster)
```

Comment:
Nit: Prefer a slightly more readable error message (and add the `xds: ` prefix). Something like

`return ClusterUpdate{}, fmt.Errorf("xds: aggregate cluster has empty clusters field in response: %+v", cluster)`

Reply:
Done. Thanks for suggestion.